braindecode/augmentation/transforms.py

```diff
@@ -0,0 +1,1000 @@
+    identity, random_bandstop, freq_shift, random_rotation,\
+    get_standard_10_20_positions
+
+
+class TimeReverse(Transform):
+    """ Flip the time axis of each feature sample with a given probability
+
+    Parameters
+    ----------
+    probability : float | callable
+        Either a float between 0 and 1 defining the uniform probability of
+        applying the operation, or a `callable` (modelling a non-uniform
+        probability law for example) taking no input and returning a boolean.
+    magnitude : object, optional
+        Always ignored, exists for compatibility.
+    mag_range : object, optional
+        Always ignored, exists for compatibility.
+    random_state: int | numpy.random.Generator, optional
+        Seed to be used to instantiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        probability,
+        magnitude=None,
+        mag_range=None,
+        random_state=None
+    ):
+        super().__init__(
+            operation=time_reverse,
+            probability=probability,
+            random_state=random_state
+        )
+
+
+class SignFlip(Transform):
+    """ Flip the sign axis of each feature sample with a given probability
+
+    Parameters
+    ----------
+    probability : float | callable
+        Either a float between 0 and 1 defining the uniform probability of
+        applying the operation, or a `callable` (modelling a non-uniform
+        probability law for example) taking no input and returning a boolean.
+    magnitude : object, optional
+        Always ignored, exists for compatibility.
+    mag_range : object, optional
+        Always ignored, exists for compatibility.
+    random_state: int | numpy.random.Generator, optional
+        Seed to be used to instantiate numpy random number generator instance.
+        Used to decide whether or not to transform given the probability
+        argument. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        probability,
+        magnitude=None,
+        mag_range=None,
+        random_state=None
+    ):
+        super().__init__(
+            operation=sign_flip,
+            probability=probability,
+            random_state=random_state
+        )
+
+
+class DownsamplingShift(Transform):
+    """ Downsamples and offsets features matrix with a given probability
+
+    Augmentation proposed in [1]_
+
+    DEPRECATED
+
+    Parameters
+    ----------
+    probability : object, optional
+        Always ignored, exists for compatibility.
+    magnitude : object, optional
+        Always ignored, exists for compatibility.
+    mag_range : object, optional
+        Always ignored, exists for compatibility.
+    factor: int, optional
+        Factor by which X should be downsampled. For example, if factor is 2,
+        only every other column of X will be kept. Defaults to 2.
```

Comment:
Not sure what you mean with "only every other column" (but this might be my bad english).
Maybe "For example, if factor is 2, this will keep one sample out of 2 in X".

Reply:
Yes, "every other" means '1 out of 2" 😝